src/java.desktop/windows/native/libawt/windows/WPrinterJob.cpp

```diff
@@ -175,6 +175,10 @@ static jobjectArray getPrinterNames(JNIEnv *env, DWORD flags) {
         }
     } catch (std::bad_alloc&) {
         delete [] pPrinterEnum;
+        if (nameArray != NULL) {
```

Comment:
Not sure that we usually clean the local refs in the native JNI methods, that only required in the native loop which are never returned to the java(I have check that by the grep of env->NewObjectArray)

Reply:
You're right. It's not required: all local refs are cleared when the JNI method exits.
My reasoning was that `env->DeleteLocalRef(utf_str)` is called in the loop. Strictly, it's not required either. Yet if there's a large number of elements in the array, it could cause a problem.

I'm going to withdraw the PR as the fix is not necessary.